### ISSUE TITLE
Linux version is not syncing 30 days of history, only current data

### ISSUE THREAD

**Issue by howels** [emericg/WatchFlower#81]
**Describe the bug**
When syncing history using Android I see many data points and a detailed graph of the past data.

When syncing on the desktop with Linux I see a single current value returned and no historical data is shown.

**Expected behavior**

Sync of history should retrieve as many data points as are currently available on the desktop as well as Android.

**To Reproduce**

Either click "sync history" on main screen or the button "synchronize history" under each sensor.

I have 4 sensors, same problem on all.

**Your environment**
Please describe the environment you are using:
1. Tested using the Flatpak on Ubuntu and Arch
2. Flathub
3. Version 5.4

**Comment by emericg** [comment]
Did you try that?
https://github.com/emericg/WatchFlower#linux-live-mode-and-historical-data

BLE on Linux is just generally not that good. Part bad BLE adapters (and very poor driver support) part just nobody really uses it... so nobody debugs it? I don't know.

**Comment by howels** [comment]
I've moved to the Home Assistant integration now, as I can more easily use multiple BLE proxies around the house.

**Comment by emericg** [comment]
I'm sorry I haven't answered before, but I wasn't working on WatchFlower and had no particular insight into the issue (linux BLE is just linux BLE...).
I just saw yesterday that I had actually written in the readme that the sync will usually not even start without that "hack", which isn't really a hack, it's just telling the kernel to not give up after only a couple errors, which is outrageous default behavior...

**Comment by howels** [comment]
The system of remote ESP32 based ble proxies with esphome and Home Assistant really increases the reliability and convenience of ble around the home. It's allowed me to get data trends over several months in multiple rooms with no issues. Much easier than trying to have a single ultra-reliable giant-antenna Linux Bluetooth adapter. Just spread several small ESP32 around and connect these to WiFi.

**Comment by emericg** [comment]
Yes the various MQTT gateways that are available for BLE or zigbee are really neat.
But I just can't with Home Assistant UI and just general UX...

**Comment by howels** [comment]
> Yes the various MQTT gateways that are available for BLE or zigbee are really neat. But I just can't with Home Assistant UI and just general UX...

True, it suffers from typical "design by committee"; a mix of many ideas without a strong visual identity. Watchflower is definitely sharper looking.
